app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -97,68 +120,138 @@ const MapImpl: React.FunctionComponent = () => {
+        if their dirty variable is true.
+
         */
         //!Object.keys(state.featureTypesFilters).includes(feature.type!))
-        for (const feature of featuresInState) {
-          const filterCounter = state.filterCounter;
-          if (
-            feature.type! === FeatureType.Transmission &&
-            filterCounter < 20
-          ) {
-            //hardcoded number of filters to add for now
-            //this is first time coming across type. Add a filter for it and create a typeRange object for it.
-            dispatch(addFilter(feature.type, feature, filterCounter));
-            return null;
+
+        for (const featureType of Object.values(FeatureType)) {
+          if (state.featuresByType[featureType].dirty) {
+            const dirtyLoaded = state.featuresByType[featureType].features;
+            //const newLoaded = [...oldLoaded, ...featuresInState];
+
+            const datasets = {
+              data: Processors.processGeojson({
+                type: "FeatureCollection",
+                features: dirtyLoaded.map(feature => {
+                  return {
+                    type: "Feature",
+                    geometry: wkt.parse(feature.geometry.wkt),
+                    properties: feature,
+                  };
+                }),
+              }),
+              info: {
+                id: featureType,
+              },
+            };
+
+            //dispatch(removeDataset(featuresInState[0].type));
+            dispatch(removeDataset(featureType));
+            dispatch(
+              addDataToMap({
+                datasets,
+                options: {centerMap: true, readOnly: true},
+              })
+            );
           }
         }
 
-        const datasets = {
-          data: Processors.processGeojson({
-            type: "FeatureCollection",
-            features: featuresInState.map(feature => {
-              return {
-                type: "Feature",
-                geometry: wkt.parse(feature.geometry.wkt),
-                properties: feature,
-              };
-            }),
-          }),
-          info: {
-            id: featuresInState[0].type,
-          },
-        };
-        dispatch(
-          addDataToMap({
-            datasets,
-            options: {centerMap: true, readOnly: true},
-          })
-        );
+        break;
+      }
 
+      /*
+      Features in this state have been rendered. In the process of them being added to the map,
+      the variable 'needsFilter' of the appropriate type has been set to true if this was the first time a 
+      feature of that type was added to the map.
+      The step here is to check what filters need to be added and call addFilter if neccessary.
+      */
+      case MapFeatureState.RENDERED: {
+        for (const featureType of Object.values(FeatureType)) {
+          if (state.featuresByType[featureType].needsFilters) {
```

Comment:
Rather than having boolean like this, I'd prefer to see another enum, MapFeatureTypeState.ADDED_FILTER or whatnot.

You recall the MapFeatureState enum was introduced so that there wouldn't be a bunch of booleans indicating what the state of something is.